src/vector.rs

```diff
@@ -224,6 +228,15 @@ impl<T: Value, N: Unsigned> ImmList<T> for VectorInner<T, N> {
     fn level_iter_from(&self, index: usize) -> LevelIter<'_, T> {
         LevelIter::from_index(index, &self.tree, self.depth, Length(N::to_usize()))
     }
+
+    fn iter_arc(&self, index: usize) -> Result<ArcIter<'_, T>, Error> {
+        Ok(ArcIter::from_index(
+            index,
+            &self.tree,
+            self.depth,
+            Length(N::to_usize()),
+        ))
```

Comment:
This should not be infallible (`ArcIter::from_index` should be able to fail if `T` is incompatible)